spock-specs/src/test/groovy/org/spockframework/smoke/mock/GroovySpies.groovy

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.spockframework.smoke.mock
+
+import org.spockframework.mock.CannotCreateMockException
+import org.spockframework.runtime.SpockException
+import org.spockframework.util.SpockDocLinks
+import spock.lang.Requires
+import spock.lang.Specification
+
+class GroovySpies extends Specification {
+  def "can spy on concrete instances"() {
+    def person = GroovySpy(new Person())
```

Comment:
Is this testing that it works? When you don't customize the instance, how would you notice the difference to the parameter being ignored?

Reply:
Afair I just ported over the tests from `JavaSpies` where it is done the same.

Reply:
Also, I'm not sure what you mean.
Calling the method on the Spy calls through to the original object.
What should be customized here?
That the Spy can intercept and change things is tested further down.

Reply:
I was talking about creating `Person` with non-default constructor values. Otherwise, it would behave the same as `GroovySpy()`.

Reply:
What would that change?
The tested method is not influenced by the constructor arguments or any property of the class.
Again, I just ported over the test from `JavaSpies`, assuming that whoever invented them had a reason to write them like that.